app/vtstorage/netinsert/netinsert.go

```diff
@@ -341,7 +342,8 @@ func (s *Storage) MustStop() {
 
 // AddRow adds the given log row into s.
 func (s *Storage) AddRow(streamHash uint64, r *logstorage.InsertRow) {
-	idx := s.srt.getNodeIdx(streamHash)
+	// todo: @jiekun the trace ID field MUST be the last field. add extra ways to secure it.
+	idx := s.srt.getNodeIdx(streamHash, r.Fields[len(r.Fields)-1].Value)
```

Comment:
Let's add extra argument to the `AddRow` function, and pass `trace_id` directly, to avoid breaking this tricky impl (use the last field as sharding key) in the future.